_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,244 @@
+---
+id: d0f69e
+name: All table header cells has corresponding data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has corresponding data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+input_aspects:
+- DOM Tree
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+This rule applies to `th` elements within a `table` that is [visible](#visible) and [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+## Expectation
+
+Each target element is either the row header or the column header for data cells that is not empty (""), within the same `table`.
+
+## Assumptions
+
+- Tables are [well-formed, according to the HTML5.1 specification](https://www.w3.org/TR/html51/tabular-data.html#forming-a-table).
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells](https://www.w3.org/TR/html/tabular-data.html#forming-relationships-between-data-cells-and-header-cells)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+`th` has corresponding `cells` within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>05:41</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 2
+
+`th` has corresponding `cells` within the same `table`. Here the cells span multiple columns.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th>Projects</th>
+      <th>Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `headers` attribute.
+
+```html
+<table>
+  <tr> 
+    <td headers="header1">
+      Projects
+    </td>
+    <th id="header1">
+      Projects
+    </th> 
+  </tr>
+</table>
+```
+
+#### Passed Example 4
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `aria-labelledby` attribute.
+
+```html
```

Comment:
You might want to correct tab spaces in this one

Reply:
This seems fine in the IDE. I have noticed GitHub diff view having issues with highlighting spacing correctly.

![image](https://user-images.githubusercontent.com/20978252/59099744-f8a20200-891b-11e9-8a65-1dfc439e2506.png)

Reply:
Yep, same here. But on Github diff it does look out of place.